kbfs: fix all outstanding golangci-lint issues, and enable during CI

strib opened the pull request:
Most of the fixes are in libkbfs, but a few snuck in in other places as well.

Issue: HOTPOT-398

strib commented:
> worthwhile to add to the precommit hook as well

@joshblum the pre-commit hooks are already present at the root of the project, right?  https://github.com/keybase/client/blob/4697d945048e109514bc4b5f5e8b305e2d7ce73c/.pre-commit-config.yaml

joshblum commented:
oh that uses https://github.com/golang/lint we should switch     `- id: golangci-lint` ( i guess stellarnet should switch over as well )

strib commented:
> oh that uses golang/lint we should switch - id: golangci-lint ( i guess stellarnet should switch over as well )

But we shouldn't add that until the whole repo is converted, right?  I don't use `pre-commit` so I don't know how it works, but seems like we shouldn't edit that root config yet.

joshblum commented:
pre-commit runs the checks on your staged changes so i think we can apply it now so new changes are checked for whoever uses that.

strib commented:
@joshblum 	ok I made the linting stricter, fixed the many new issues that brought about, and added `golangci-lint` to the pre-commit config.  Please take another look!

jakob223 commented on go/kbfs/test/engine_common.go:
```diff
@@ -42,8 +42,8 @@ func setBlockSizes(t testing.TB, config libkbfs.Config, blockSize, blockChangeSi
 	if bwKBps > 0 {
-		config.SetBlockOps(libkbfs.NewBlockOpsConstrained(
-			config.BlockOps(), bwKBps))
+		// TODO: reinstate constrained bandwidth tests.
```
Was this breaking as a result of the lint changes or something? I'm confused.

strib replied:
Turns out this broke a long time ago (in the days of the old `kbfs` repo) on a refactor of the `BlockOps` functions.  But these constrained benchmark tests are either run manually (and probably haven't even been run since that happened) or only used as a crutch to set the background flushes flag, so it didn't seem worth it to try to fix as part of this PR.

strib commented on go/kbfs/data/file_data.go:
```diff
@@ -976,8 +976,9 @@ func (fd *FileData) Split(ctx context.Context, id tlf.ID,
 				iptrs := pblock.IPtrs
-				pblock.IPtrs =
-					append(iptrs[:pb.childIndex], iptrs[pb.childIndex+1:]...)
+				pblock.IPtrs = make([]IndirectFilePtr, len(iptrs)-1)
```
The linter complains when you assign the append result to a different variable than the first parameter to append.  Presumably because of the weird interactions with append and memory allocation?

strib commented on go/kbfs/libkbfs/bserver_remote.go:
```diff
@@ -415,7 +409,7 @@ func (b *BlockServerRemote) Get(
 				// the `Get`. But we need this cached synchronously so prefetch
 				// operations can work correctly.
-				dbc.Put(ctx, tlfID, id, buf, serverHalf, cacheType)
+				_ = dbc.Put(ctx, tlfID, id, buf, serverHalf, cacheType)
```
The disk block cache already logs it, so no need to do it here too.  I'll add a comment.

strib commented on go/kbfs/libkbfs/config_local.go:
```diff
@@ -259,7 +259,7 @@ func NewConfigLocal(mode InitMode,
 	config.SetCodec(kbfscodec.NewMsgpack())
 	if diskCacheMode == DiskCacheModeLocal {
-		config.loadSyncedTlfsLocked()
+		_ = config.loadSyncedTlfsLocked()
```
Also logged in the function itself.

strib commented on go/kbfs/libkbfs/disk_block_cache_wrapped.go:
```diff
@@ -173,7 +173,7 @@ func (cache *diskBlockCacheWrapped) moveBetweenCachesWithBlockLocked(
 	go func() {
 		defer cache.deleteGroup.Done()
-		secondaryCache.Delete(ctx, []kbfsblock.ID{blockID})
+		_, _, _ = secondaryCache.Delete(ctx, []kbfsblock.ID{blockID})
```
This is a "best effort" kind of delete, the block might not even be in there.  I'll add a comment.
